### PR TITLE
Add `ModuleType` as a possible type to `URLResolver.urlconf_name`

### PR DESCRIPTION
**Pull request by Viicos**
According to code comment:
It may also be an object with an urlpatterns attribute (i.e. a module object)

# I have made things!

### CODE REVIEW

File: django-stubs/urls/resolvers.pyi [hunk 1]
```diff
@@ -103,7 +103,7 @@ class URLPattern:
 
 class URLResolver:
     pattern: _Pattern
-    urlconf_name: str | None | Sequence[_AnyURL]
+    urlconf_name: str | None | Sequence[_AnyURL] | ModuleType
```
Comment (sobolevn) [inline django-stubs/urls/resolvers.pyi]: Do we need to update the `__init__` type as well?

Comment (Viicos) [reply django-stubs/urls/resolvers.pyi]: Yes, it seems to be set in the `__init__` [here](https://github.com/django/django/blob/14917c9ae272f47d23401100faa6cefa8e1728bf/django/urls/conf.py#L74)

**Review by sobolevn — APPROVED** [review]
Seems correct to me: https://github.com/django/django/blob/14917c9ae272f47d23401100faa6cefa8e1728bf/django/urls/resolvers.py#L725-L730

Thanks!

**Review by sobolevn — CHANGES_REQUESTED** [review]
Wait a second, can it be `None`? 🤔 
Looks like https://github.com/django/django/blob/14917c9ae272f47d23401100faa6cefa8e1728bf/django/urls/resolvers.py#L483-L486 does not agree with this

**Comment by Viicos** [comment]
> Wait a second, can it be `None`? 🤔
> Looks like https://github.com/django/django/blob/14917c9ae272f47d23401100faa6cefa8e1728bf/django/urls/resolvers.py#L483-L486 does not agree with this

True, [this search](https://github.com/search?q=repo%3Adjango%2Fdjango%20%22URLResolver(%22&type=code) doesn't show any case where it is `None` (at least in the Django codebase)

**Review by sobolevn — APPROVED** [review]
Thank you!

**Pull request merged**